### PR TITLE
Added functionality for the icons of the map control to make it overrideable on the theme

### PR DESCRIPTION
**Pull request by nicolasmartinrojo**

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ export {
   SaveExportDropdownFactory,
   PanelHeaderDropdownFactory
 } from './side-panel/panel-header';
-export {default as PanelHeaderAction} from './side-panel/panel-header-action';
+export {default as PanelHeaderActionFactory} from './side-panel/panel-header-action';
```
Comment (heshan0131) [inline src/components/index.js:48]: Still need to export `PanelHeaderAction`

File: src/components/map/map-control.js [hunk 2]
```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
...
-    <MapControlTooltip
-      id="action-toggle"
-      message={isSplit ? 'tooltip.closePanel' : 'tooltip.switchToDualView'}
-    />
-  </MapControlButton>
-));
+MapLegendPanelFactory.deps = [];
+export function MapLegendPanelFactory() {
+  const defaultActionIcons = {
+    legend() {
```
Comment (heshan0131) [inline src/components/map/map-control.js:185]: make it a key value object
```
const defaultActionIcons = {
    legend: () => <Legend height="22px" />;
}
```

File: src/components/map/map-control.js [hunk 3]
```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
...
+        key={2}
+        data-tip
+        data-for="show-legend"
+        className="map-control-button show-legend"
+        onClick={e => {
+          e.preventDefault();
+          onToggleMenuPanel();
+        }}
+      >
+        {actionIcons.legend()}
```
Comment (heshan0131) [inline src/components/map/map-control.js:210]: {<actionIcons.legent/>}

File: src/components/map/map-control.js [hunk 4]
```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
...
-      message={dragRotate ? 'tooltip.disable3DMap' : 'tooltip.3DMap'}
-    />
-  </MapControlButton>
-));
+  MapLegendPanel.displayName = 'MapControlPanel';
+  return MapLegendPanel;
+}
+SplitMapButtonFactory.deps = [];
+export function SplitMapButtonFactory() {
+  const defaultActionIcons = {
```
Comment (heshan0131) [inline src/components/map/map-control.js:231]: same here

File: src/components/map/map-control.js [hunk 5]
```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
...
+    )
+  );
 
-Toggle3dButton.displayName = 'Toggle3dButton';
+  SplitMapButton.displayName = 'SplitMapButton';
+  return SplitMapButton;
+}
+Toggle3dButtonFactory.deps = [];
+export function Toggle3dButtonFactory() {
+  const defaultActionIcons = {
```
Comment (heshan0131) [inline src/components/map/map-control.js:266]: same

File: src/components/map/map-control.js [hunk 6]
```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
...
-      </div>
-    );
-  }
-);
+MapDrawPanelFactory.deps = [];
+export function MapDrawPanelFactory() {
+  const defaultActionIcons = {
+    visible: EyeSeen,
+    hidden: EyeUnseen,
+    polygon() {
```
Comment (heshan0131) [inline src/components/map/map-control.js:304]: each item should have the same construct, why is value of `visible` a component, and `polygon` a function?
if you are providing a prop API does it make sense for people to provide icons in different type?

```
polygon: () => <DrawPolygon height="22px" />
```

File: src/components/index.js (hunk 1 above)
Comment (nicolasmartinrojo) [reply src/components/index.js:49]: Fixed

File: src/components/map/map-control.js (hunk 2 above)
Comment (nicolasmartinrojo) [reply src/components/map/map-control.js]: fixed

File: src/components/map/map-control.js (hunk 3 above)
Comment (nicolasmartinrojo) [reply src/components/map/map-control.js]: fixed, good catch

File: src/components/map/map-control.js (hunk 4 above)
Comment (nicolasmartinrojo) [reply src/components/map/map-control.js:229]: fixed

File: src/components/map/map-control.js (hunk 5 above)
Comment (nicolasmartinrojo) [reply src/components/map/map-control.js:260]: fixed

File: src/components/map/map-control.js (hunk 6 above)
Comment (nicolasmartinrojo) [reply src/components/map/map-control.js]: you're correct. fixed

**Review by heshan0131 — APPROVED** [review]

**Pull request merged**